### PR TITLE
Merge mission sync

### PR DESCRIPTION
**Pull request by huachen24**
This pull request adds 2 methods of file syncing to OGC
1. Direct method where the GCS and a specified plane's waypoint folders are synced
2. Functions to sync GCS with server and server with plane
Will require RQT to integrate these functions

### CODE REVIEW

File: ogc_ws/src/despatcher/scripts/missionserver.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+import os
+import rospy
+from pathlib import Path
+import paramiko
+import time
+
+home_dir = str(Path.home())
+serverfolder = "/home/ubuntu/Waypoints/"
```
Comment (rumesh986) [inline ogc_ws/src/despatcher/scripts/missionserver.py]: Move the home_dir declaration inside the init?

File: ogc_ws/src/despatcher/scripts/g2a.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+			self._statustext_flag = False
+		else:
+			return
+	else:
+		return
+	self._send_ack()
+
+def check_arming(self):
+	"""Check for Arm/Disarm commands from Ground Control"""
+	rospy.wait_for_service('mavros/cmd/arming')
```
Comment (rumesh986) [inline ogc_ws/src/despatcher/scripts/g2a.py]: Should probably move these back into the init function of the class

File: ogc_ws/src/despatcher/scripts/missionserver.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import paramiko
+import time
+
+serverfolder = "/home/ubuntu/Waypoints/"
+
+class MissionServer():
+	def __init__(self):
+		home_dir = str(Path.home())
+		server_user = "ubuntu"
+		server_ip = "46.137.242.90"
```
Comment (disgruntled-patzer) [inline ogc_ws/src/despatcher/scripts/missionserver.py:16]: We'll keep this first, but eventually the following must be done:

1. The server ip must be broken out into the identifiers file (and I'll sync it with the way sbd handles it)
2. Remove the private key and use public key authorization

File: ogc_ws/src/despatcher/src/air_despatcher.py [hunk 4]
```diff
@@ -298,22 +112,32 @@ def check_incoming_msgs(self, data):
...
+                g2a.check_mode(self)
             elif "wp" in self._recv_msg or "mission" in self._recv_msg:
-                self._check_mission()
+                g2a.check_mission(self)
         except(rospy.ServiceException):
             rospy.logwarn("Service Call Failed")
         except (ValueError, IndexError):
             rospy.logerr("Invalid message format")
 
+    def _push_waypoint(self, waypoints):
```
Comment (disgruntled-patzer) [inline ogc_ws/src/despatcher/src/air_despatcher.py]: Duplicate function from waypoint.py?

**Review by disgruntled-patzer — CHANGES_REQUESTED** [review]
Remove the duplicate function and we should be good to go

**Pull request merged**